doc/type.rst

```diff
@@ -455,22 +405,87 @@ Now the type is created, the SearchFactory needs to know it exists.
 
 And then registering at the FactoryBuilder.
 
-.. note::
-
-    Because the InvoiceNumberType uses the 'constraints' options the
-    validator extension must be enabled as well.
-
 .. code-block:: php
     :linenos:
 
     /* ... */
 
-    use Rollerworks\Component\Search\Extension\Validator\ValidatorExtension;
     use Acme\Invoice\Search\Extension\Invoice\InvoiceExtension;
 
     $searchFactory = new Searches::createSearchFactoryBuilder()
-        ->addExtension(new ValidatorExtension())
         ->addExtension(new InvoiceExtension())
         ->getSearchFactory();
 
 That's it the type is now ready for usage.
+
+Testing Field Types
+-------------------
+
+Now that you have successfully created you're first field type
+its a good idea to test if it actually does what you except.
+
+Fortunately RollerworksSearch comes with a very handy PHPUnit Test class
+to help you with testing a field type.
+
+.. note::
+
+    You're tests should test if correctly formatted input produces the right result.
+    But they should also test that incorrectly formatted (or invalid) input fails to transform
+    and is not accepted!
+
+Continuing from the InvoiceNumberType field type above;
+
+.. code-block:: php
+    :linenos:
+
+    use Rollerworks\Component\Search\Test\FieldTypeTestCase;
+    use Acme\Invoice\Search\Extension\Invoice\InvoiceExtension;
+    use Acme\Invoice\InvoiceNumber;
+
+    class InvoiceNumberTypeTest extends FieldTypeTestCase
+    {
+        public function testValidInvoiceNumber()
+        {
+            $field = $this->getFactory()->createField('invoice', 'invoice_number');
+
+            $expectedOutput = new InvoiceNumber(2015, 20);
+            $expectedView = '2015-0020';
+
+            $this->assertTransformedEquals($field, $expectedOutput, '2015-0020', $expectedView);
+            $this->assertTransformedEquals($field, $expectedOutput, '2015-020', $expectedView);
+            $this->assertTransformedEquals($field, $expectedOutput, '2015-20', $expectedView);
+        }
+
+        public function testWrongInputFails()
+        {
+           $field = $this->getFactory()->createField('invoice', 'invoice_number');
+
+            $this->assertTransformedFails($field, '201-0020');
+            $this->assertTransformedFails($field, '2015-');
+            $this->assertTransformedFails($field, '201500');
+        }
+
+        protected function getTestedType()
+        {
+            return 'invoice_number';
+        }
+
+        protected function getExtensions()
+        {
+            return array(new InvoiceExtension());
+        }
+
+        /* If you don't use a SearchExtension use the following instead */
+
+        protected function getTypes()
+        {
+            return array(
```

Comment:
I have not considered that :) This library has existed for a long time (before PHP 5.4 was even out).
I currently don't have any special cases like traits so I'm not sure if bumping the version requirements is a good idea.